web/packages/core/src/swf-file-name.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Create a filename to save a downloaded SWF into.
+ *
+ * @param swfUrl The URL of the SWF file.
+ * @returns The filename the SWF file can be saved at.
+ */
+export function swfFileName(swfUrl: string): string {
+    let name = swfUrl.substring(swfUrl.lastIndexOf("/") + 1);
```

Comment:
maybe instead of manual string manipulation, we could use https://developer.mozilla.org/en-US/docs/Web/API/URL ? Should be a bit easier this way.

Reply:
elsewhere in `ruffle-player.ts` there is
```
try {
    this.swfUrl = new URL(
        options.url,
        document.location.href
    ).href;
} catch {
    this.swfUrl = options.url;
}
```

I'm not sure when the catch block would get hit, but it looks like the code was done defensively in case `swfUrl` wound up not being a valid URL. Not sure when that would happen. Page overriding `URL.prototype`?

Reply:
Looks like that was a suggestion from Mike: https://github.com/ruffle-rs/ruffle/pull/1957#discussion_r542053627

Reply:
+1 for using `URL` over manual string manipulation. I'm not sure how the `catch` flow is useful at all, because invalid URLs will later fail in `WebNavigatorBackend::fetch` when trying to load the root movie. So I suggest removing the `catch` case (but be sure a descriptive error message is still shown), and making `swfUrl` of type `URL`, such that the various components (`pathname` etc.) can be accessed without an additional `URL` parsing. This will also save the `new URL()` in `displayRootMovieDownloadFailedMessage` along the way.

And a page overriding `URL.prototype` could also override anything else (e.g. `String.prototype`), so we have to deal with it anyway.

Reply:
> invalid URLs will later fail in WebNavigatorBackend::fetch

unless the page overrides `fetch`. I know the internet archive does; I'll check what swfUrls it passes. I imagine they're valid, though.

Reply:
checked IA: the URL is valid, IA just prepends the archive.org URL to the front.

Reply:
So I still stand behind my initial suggestion.

Reply:
I did that (for getting rid of the query string), but I still had to do manual string manipulation for getting everything after the last slash.